File: server/CinemasService/Errors/LayoutErrors.cs
namespace CinemasService.Errors
{
    public static class LayoutErrors
    {
        public static class Create
        {
            public const string NameRequired = "name_required";
            public const string NameMaxLength = "name_max_length";
            public const string NameAlreadyExists = "name_already_exists";

            public const string RowsRequired = "rows_required";
            public const string ColumnsRequired= "columns_required";

            public const string ElementsRequired = "elements_requried";

            public const string ElementsTooMany = "elements_too_many";

            public const string SeatLabelRequired = "seat_label_required";
            public const string SeatLabelDuplicated = "seat_label_duplicated";
            public const string DimensionsTooSmall = "dimensions_too_small";
            public const string NotEnoughSeats = "not_enough_seats";
            public const string DuplicatePositions = "duplicate_positions";
            public const string MissingSeatPositions = "missing_seat_positions";

        }

        public static class Update
        {
            public const string LayoutNotFound = "layout_not_found";

        }

        public static class Get
        {
            public const string LayoutNotFound = "layout_not_found";

        }

        public static class Delete
        {
            public const string LayoutNotFound = "layout_not_found";
        }

    }
}

File: ui/cms/src/validation/api-errors/LayoutErrors.ts
export const LayoutErrors: Record<string, string> = {
  name_required:
    "El nombre es obligatorio. Por favor, ingrese un nombre para el layout.",
  name_max_length:
    "El nombre es demasiado largo. El máximo permitido es 100 caracteres.",
  name_already_exists:
    "Ya existe un layout con este nombre. Por favor, elija otro nombre.",

  elements_required:
    "Debe incluir elementos en el layout. No puede estar vacío.",
  elements_too_many:
    "El layout tiene demasiados elementos. El máximo permitido es 500.",

  seat_label_required: "Cada asiento debe tener una etiqueta identificadora.",
  seat_label_duplicated:
    "Hay etiquetas de asiento duplicadas. Cada asiento debe tener una etiqueta única.",

  dimensions_too_small:
    "Las dimensiones son demasiado pequeñas. Mínimo requerido: 5 filas × 5 columnas.",
  not_enough_seats:
    "No hay suficientes asientos. Se requieren al menos 10 asientos válidos.",

  duplicate_positions:
    "Hay posiciones duplicadas en el layout. Cada asiento debe tener una posición única.",
  missing_seat_positions:
    "Faltan posiciones de asientos. Complete todas las posiciones requeridas.",
};
